cmd/startup/startup.go

```diff
@@ -0,0 +1,308 @@
+package main
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"flag"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/api/validate"
+	"github.com/openshift/openshift-azure/pkg/arm"
+	"github.com/openshift/openshift-azure/pkg/cluster"
+	"github.com/openshift/openshift-azure/pkg/tls"
+	azureclientstorage "github.com/openshift/openshift-azure/pkg/util/azureclient/storage"
+	"github.com/openshift/openshift-azure/pkg/util/configblob"
+	"github.com/openshift/openshift-azure/pkg/util/log"
+	"github.com/openshift/openshift-azure/pkg/util/template"
+)
+
+var (
+	logLevel         = flag.String("loglevel", "Debug", "valid values are Debug, Info, Warning, Error")
+	runningUnderTest = flag.Bool("runningundertest", false, "TestConfig.RunningUnderTest")
+	dnsHostname      = flag.String("dnshostname", "", "hostname -d")
+	gitCommit        = "unknown"
+)
+
+const (
+	origin = "/etc/origin"
+	etcd   = "/etc/etcd"
+)
+
+type startup struct {
+	blob azureclientstorage.Blob
+	log  *logrus.Entry
+}
+
+func (s *startup) writeKubeconfigs(cs *api.OpenShiftManagedCluster) error {
```

Comment:
Am wondering about the following: follow a pattern of putting `{{ YamlMarshal .Config.AdminKubeconfig | Base64Encode }}` in a file /pkg/arm/data/etc/origin/node/node.kubeconfig and get rid of writeKubeconfigs().  The equivalent is possible to get rid of writeCerts.  In writeTemplatedFiles, check if the filename has suffix ".kubeconfig"/".key" to control using the file mode 0600.

Reply:
interesting, I'll give that a go